Could not find module ‘GHC.IntWord64’ on GHC 9.2 + i386

As reported by @mpickering at https://gitlab.haskell.org/ghc/ghc/-/issues/20781#note_407377:

> On the 9.2 branch the bump seems to fail to build on the i386 job.
> https://gitlab.haskell.org/ghc/ghc/-/jobs/937122
> ```
> libraries/bytestring/Data/ByteString/Builder/RealFloat/Internal.hs:90:8: error:
>    Could not find module ‘GHC.IntWord64’
>    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
>    |
>90 | import GHC.IntWord64
>    |        ^^^^^^^^^^^^^
> make[1]: *** [libraries/bytestring/dist-install/build/.depend-v-dyn.haskell] Error 1
> ```

--- a/Data/ByteString/Builder/RealFloat/Internal.hs
+++ b/Data/ByteString/Builder/RealFloat/Internal.hs
@@ -86,7 +86,7 @@ import qualified Foreign.Storable as S (poke)
 #include <ghcautoconf.h>
 #include "MachDeps.h"
 
-#if WORD_SIZE_IN_BITS < 64 && __GLASGOW_HASKELL__ < 901
+#if WORD_SIZE_IN_BITS < 64 && !MIN_VERSION_ghc_prim(0,8,0)
 import GHC.IntWord64
 #endif
 

--- a/Data/ByteString/Builder/RealFloat/Internal.hs
+++ b/Data/ByteString/Builder/RealFloat/Internal.hs
@@ -86,7 +86,7 @@ import qualified Foreign.Storable as S (poke)
 #include <ghcautoconf.h>
 #include "MachDeps.h"
 
-#if WORD_SIZE_IN_BITS < 64 && __GLASGOW_HASKELL__ < 903
+#if WORD_SIZE_IN_BITS < 64 && __GLASGOW_HASKELL__ < 901
 import GHC.IntWord64
 #endif
 

